### PR TITLE
Weak_form for O/P differential equation/sytem.

### PR DESCRIPTION
**Pull request by aminevdom**
1) Method weak_loss is added in Solution class.
2) New examples with weak form are created.
3) Solution print is fixed for system
4) weak_form parameter is added in Solver class for weak loss eval-n

### CODE REVIEW

File: examples/example_Lotka_Volterra_paper.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    
+  
+    end_loss = Solution(grid, equation, model, 'NN').loss_evaluation(lambda_bound=100)
+    exp_dict_list.append({'grid_res':grid_res,'time':end - start,'RMSE':error_rmse.detach().numpy(),'loss':end_loss.detach().numpy(),'type':'Lotka_eqn','cache':CACHE})
+    
+    print('Time taken {}= {}'.format(grid_res, end - start))
+    print('RMSE {}= {}'.format(grid_res, error_rmse))
+    print('loss {}= {}'.format(grid_res, end_loss))
+
+    plt.figure()
```
Comment (SuperSashka) [inline examples/example_Lotka_Volterra_paper.py:174]: Сам скрипт-то рабочий, но на сетке 50 узлов кажется решения-то не получается.

File: examples/example_weak_Schrodinger.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+        torch.nn.Linear(100, 2)
+    )
+def v(grid):
+    return torch.sin(grid[:,0])+torch.sin(2*grid[:,0])+torch.sin(3*grid[:,0])+grid[:,1]
+
+weak_form=[v]
+
+equation = Equation(grid, schrodinger_eq, bconds).set_strategy('NN')
+
+model = Solver(grid, equation, model, 'NN', weak_form=weak_form).solve(lambda_bound=1, verbose=1, learning_rate=0.8,
```
Comment (SuperSashka) [inline examples/example_weak_Schrodinger.py:205]: Тут кажется решает фигово. Судя по всему надо подбирать функцию.

**Review by SuperSashka — APPROVED** [review]
Я думаю, что можно вливать, со Шрёдингером разобрались в обычной постановке, но не в слабой, надо подумать - всё остальное более-менее работает.
